### PR TITLE
[BACKLOG-30667] - CSRF - /api/scheduler changes

### PR DESCRIPTION
**Pull request by sVarma1992**
[Backlog 30667] -CSRF - /api/scheduler changes

### CODE REVIEW

File: assemblies/pentaho-war/src/main/webapp/js/parameters.js [hunk 1]
```diff
@@ -280,12 +284,16 @@ function doPost( url, query, func) {
...
+  var protectedUrl = getPentahoBaseUrl() + url;
+  var csrfToken = pho.csrfUtil.getToken(protectedUrl);
   // submit the request
   http_request.open('POST', url, true);
   http_request.setRequestHeader("Content-type", "application/json");
   http_request.setRequestHeader("Content-length", query.length);
   http_request.setRequestHeader("Connection", "close");
+  if(csrfToken !== null) {
+             xhr.setRequestHeader(csrfToken.header, csrfToken.token);
+  }
```
Comment (dcleao) [inline assemblies/pentaho-war/src/main/webapp/js/parameters.js:296]: Formatting is off (indentation).

File: assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-csrf.xml [hunk 2]
```diff
@@ -64,6 +64,25 @@
...
 
+        <!--
+          POST /api/scheduler/job
+        -->
+        <swm:regex-request-matcher pattern="^/api/scheduler/job\b.*" methods="POST" />
+
+        <!--
+          POST /api/scheduler/job/update
+        -->
+        <swm:regex-request-matcher pattern="^/api/scheduler/job/update\b.*" methods="POST" />
```
Comment (dcleao) [inline assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-csrf.xml:75]: I believe the regular expression of the first request matcher "shadows" that of the second.
You could use the first only, and, in the XML comment, put the two URLs that would match.
I checked and there appear to be no other endpoints in the `/api/scheduler` space that would match.

**Review by dcleao — CHANGES_REQUESTED** [review]
We're almost there!

File: assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-csrf.xml (hunk 2 above)
Comment (sVarma1992) [reply assemblies/pentaho-solutions/src/main/resources/pentaho-solutions/system/applicationContext-spring-security-csrf.xml]: changed as requested

File: assemblies/pentaho-war/src/main/webapp/js/parameters.js (hunk 1 above)
Comment (sVarma1992) [reply assemblies/pentaho-war/src/main/webapp/js/parameters.js]: changed as expected

**Review by dcleao — APPROVED** [review]

**Comment by dcleao** [comment]
Merge with:
- https://github.com/pentaho/pentaho-kettle/pull/8223
- https://github.com/webdetails/cdf/pull/1138

**Pull request closed without merging**